### PR TITLE
combine secrets verrazzano-cluster-elasticsearc and verrazzano-cluster-registration

### PR DESCRIPTION
**Pull request by guoyongzhang**
# Description

combine secrets verrazzano-cluster-elasticsearc and verrazzano-cluster-registration

Fixes VZ-2378

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go [hunk 1]
```diff
@@ -224,12 +223,6 @@ func TestReconcileCreateCoherenceWithLogging(t *testing.T) {
 	cli.EXPECT().
```
Comment (mgianatagh) [inline application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:223]: On line 221 should be comment be that the get is for the verrazzano-cluster-registration secret?

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go [hunk 2]
```diff
@@ -224,12 +223,6 @@ func TestReconcileCreateCoherenceWithLogging(t *testing.T) {
...
 		Get(gomock.Any(), testESSecretFullName, gomock.Not(gomock.Nil())).
```
Comment (mgianatagh) [inline application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:224]: Rename second parameter to something like testClusterRegSecretFullName?

File: application-operator/controllers/helidonworkload/helidonworkload_controller_test.go [hunk 3]
```diff
@@ -314,12 +313,6 @@ func TestReconcileCreateVerrazzanoHelidonWorkloadWithLoggingScope(t *testing.T)
 			return nil
 		}).Times(1)
 
-	// needs cluster name, expect a call to get verrazzano-cluster secret
-	cli.EXPECT().
-		Get(gomock.Any(), clusters.MCRegistrationSecretFullName, gomock.Not(gomock.Nil())).
-		DoAndReturn(func(ctx context.Context, name types.NamespacedName, sec *v1.Secret) error {
-			return nil
-		})
 	// expect a call to get the elasticsearch secret in app namespace - return not found
```
Comment (mgianatagh) [inline application-operator/controllers/helidonworkload/helidonworkload_controller_test.go:316]: similar comment as before - this secret is now a verrazzano-cluster-registration secret

File: application-operator/controllers/loggingscope/fluentd.go [hunk 4]
```diff
@@ -535,14 +545,13 @@ func createEmptySecretForFluentdVolume(ctx context.Context, cli k8sclient.Client
 
 // copies the managed cluster Elasticsearch secret to the given namespace/name IF it exists
 func copyManagedClusterVMISecret(ctx context.Context, cli k8sclient.Client, namespace string, name string) error {
```
Comment (mgianatagh) [inline application-operator/controllers/loggingscope/fluentd.go:547]: copying the cluster registration secret

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go (hunk 1 above)
Comment (guoyongzhang) [reply application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:223]: updated

File: application-operator/controllers/helidonworkload/helidonworkload_controller_test.go (hunk 3 above)
Comment (guoyongzhang) [reply application-operator/controllers/helidonworkload/helidonworkload_controller_test.go:316]: updated

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go (hunk 2 above)
Comment (guoyongzhang) [reply application-operator/controllers/cohworkload/coherenceworkload_controller_test.go]: updated

File: application-operator/controllers/loggingscope/fluentd.go (hunk 4 above)
Comment (guoyongzhang) [reply application-operator/controllers/loggingscope/fluentd.go]: It now only copies the selected fields.

File: application-operator/constants/constants.go [hunk 5]
```diff
@@ -29,22 +29,22 @@ const AdminKubeconfigData = "admin-kubeconfig"
...
-// ElasticsearchURLData - the field name in ElasticsearchSecret that contains the admin cluster's
+// ElasticsearchURLData - the field name in MCRegistrationSecret that contains the admin cluster's
 // Elasticsearch endpoint's URL
-const ElasticsearchURLData = "url"
+const ElasticsearchURLData = "es-url"
 
-// ElasticsearchUsernameData - the field name in ElasticsearchSecret that contains the admin
+// ElasticsearchUsernameData - the field name in MCRegistrationSecret that contains the admin
 // cluster's Elasticsearch username
 const ElasticsearchUsernameData = "username"
```
Comment (mgianatagh) [inline application-operator/constants/constants.go:38]: should this field name be pre-fixed with "es-"

File: application-operator/constants/constants.go [hunk 6]
```diff
@@ -29,22 +29,22 @@ const AdminKubeconfigData = "admin-kubeconfig"
...
 
-// ElasticsearchPasswordData - the field name in ElasticsearchSecret that contains the admin
+// ElasticsearchPasswordData - the field name in MCRegistrationSecret that contains the admin
 // cluster's Elasticsearch password
 const ElasticsearchPasswordData = "password"
```
Comment (mgianatagh) [inline application-operator/constants/constants.go:42]: should this be prefix with "es-"

File: application-operator/constants/constants.go [hunk 7]
```diff
@@ -29,22 +29,22 @@ const AdminKubeconfigData = "admin-kubeconfig"
...
 
+// ElasticsearchCABundleData - the field name in MCRegistrationSecret that contains the admin
+// cluster's Elasticsearch CA bundle
+const ElasticsearchCABundleData = "ca-bundle"
```
Comment (mgianatagh) [inline application-operator/constants/constants.go:46]: should this be prefixed with "es-"

File: application-operator/constants/constants.go (hunk 5 above)
Comment (guoyongzhang) [reply application-operator/constants/constants.go:38]: "es-" would be nice.  But it is a lot of work.  Maybe file a JIRA for that?

File: application-operator/constants/constants.go (hunk 6 above)
Comment (guoyongzhang) [reply application-operator/constants/constants.go:42]: "es-" would be nice.  But it is a lot of work for it.  Maybe file a JIRA for that?

File: application-operator/constants/constants.go (hunk 7 above)
Comment (guoyongzhang) [reply application-operator/constants/constants.go:46]: "es-" would be nice.  But it is a lot of work for it.  Maybe file a JIRA for that?

**Review by mgianatagh — APPROVED** [review]

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**